src/components/MarkdownEditor.vue

```diff
@@ -0,0 +1,97 @@
+<script setup lang="ts">
+defineProps<{
+  modelValue: string;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:modelValue', value: string);
+}>();
+
+const editorContainerRef = ref<HTMLDivElement | null>(null);
+const editorFileInputRef = ref<HTMLInputElement | null>(null);
+const editorRef = ref<HTMLTextAreaElement | null>(null);
+const editor = useMarkdownEditor(editorRef, editorFileInputRef, editorContainerRef, value => {
+  emit('update:modelValue', value);
+});
+
+const previewEnabled = ref(false);
+</script>
+
+<template>
+  <div class="flex">
+    <Link :is-active="!previewEnabled" text="Write" class="pr-3" @click="previewEnabled = false" />
+    <Link :is-active="previewEnabled" text="Preview" @click="previewEnabled = true" />
+  </div>
```

Comment:
Can we just move this out of the component?

Reply:
Just elements to switch between two modes (and passing previewEnabled as prop to display preview within this component), or entire preview functionality (so consumer of this component will need its own preview component)?

Reply:
I mean the entire preview functionality